### PR TITLE
Dynamic popup

### PR DESCRIPTION
**Pull request by Lul-Abdifan**
**In this milestone project,We have done;-**
-Use javascript array to store objects
-Update to dynamically generated cards for mobile version
-Update to dynamically generated cards for desktop version
-Add popup for window and mobile version
-Add blur effect for popup background
-Fix eslint error

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -134,13 +139,13 @@ <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
...
          </div>
 
-          <!--Card3-->
+  
 
-         <div class="cards card3" >
+         <!-- <div class="cards card3" >
             <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
              <div class="cards_description">
                <p>A daily selection of privately personalized reads; no accounts or sign-ups
```
Comment (leonardodiasb) [inline index.html]: - [ ] Please remove commented code. This makes your code look messy.

File: index.html [hunk 2]
```diff
@@ -157,11 +162,11 @@ <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
...
-               <a href="#"><button type="button" id="button_linkProj">See Project</button></a>  
+               <button type="button" id="button_linkProj" class="btn2">See Project</button>
              </div>
          </div>
 
-          <!--Card4-->
+     
 
             
          <div class="cards card4" >
```
Comment (leonardodiasb) [inline index.html]: - [ ] Please remove commented code. This makes your code look messy.

File: index.html [hunk 3]
```diff
@@ -181,11 +186,10 @@ <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
...
              <div class="seeProjectCards">
-               <a href="#"><button type="button" id="button_linkProj">See Project</button></a>  
+               <button type="button" id="button_linkProj" class="btn3">See Project</button>
              </div>
          </div>
 
-         <!--Card5-->
 
          <div class="cards card5" >
             <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
```
Comment (leonardodiasb) [inline index.html]: - [ ] Please remove commented code. This makes your code look messy.

File: index.html [hunk 4]
```diff
@@ -204,12 +208,10 @@ <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
...
+              <button type="button" id="button_linkProj" class="btn4">See Project</button>  
              </div>
          </div>
-
-         <!--Card6-->
-            
+          
 
             <div class="cards card6" >
                 <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
```
Comment (leonardodiasb) [inline index.html]: - [ ] Please remove commented code. This makes your code look messy.

File: index.html [hunk 5]
```diff
@@ -228,11 +230,10 @@ <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
...
                  <div class="seeProjectCards">
-                   <a href="#"><button type="button" id="button_linkProj">See Project</button></a>  
+                   <button type="button" id="button_linkProj" class="btn5">See Project</button>
                  </div>
              </div>
 
-          <!--Card7-->
       
              <div class="cards card7">
                 <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
```
Comment (leonardodiasb) [inline index.html]: - [ ] Please remove commented code. This makes your code look messy.

File: index.html [hunk 6]
```diff
@@ -251,9 +252,9 @@ <h3 class="cards-header">Profesional Art <br>Printing Data</h3>
                  </ul>
                </div>
                  <div class="seeProjectCards">
-                   <a href="#"><button type="button" id="button_linkProj">See Project</button></a>  
+                  <button type="button" id="button_linkProj" class="btn6">See Project</button> 
                  </div>
-             </div>
+             </div> -->
```
Comment (leonardodiasb) [inline index.html]: - [ ] Please remove commented code. This makes your code look messy.

File: script.js [hunk 7]
```diff
@@ -18,3 +18,149 @@ function close() {
...
+document.querySelectorAll('#button_linkProj').forEach((btn) => {
+  btn.onclick = () => {
+    popup.style.display = 'block';
+    document.body.style.position = 'fixed';
+    forbur.classList.add('blur');
+
+    projects.forEach((project) => {
+      if (btn.className === project.id) {
+        popup.innerHTML = render(project);
+      }
```
Comment (leonardodiasb) [inline script.js:202]: - [x] Your function for rendering the popup window is rendering the same popup for all the projects. The popup window should display the information from the card that the user clicked. Kindly fix this issue so the user can get the correct information.

**Review by leonardodiasb — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi @Lul-Abdifan, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 

 - No linter errors. ✔️

## Aspects to improve ♻️

- [x] Check the comments under the review.
- [x] Your work section should be all refactored and created dynamically with Javascript. You left 2 projects hard-coded. Please make those projects dynamically created with JS as you did with the others projects.
- [x] The second project you added is the hover version of the project card in the desktop version. When the user hovers, the project card should change to that appearance and the 'See Project' button should appear so the user can open the popup window. Kindly add that feature so the user can see the popup of the respective project. There is no way the user can access the popup window in the desktop version of your version.
-Your desktop version:
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 1 above)
Comment (Lul-Abdifan) [reply index.html]: @leonardodiasb  Thank you for your feedback.I will do on my issues :+1: Happy Coding :smile:

**Comment by Lul-Abdifan** [comment]
**Required changes are Done** :smile:
 - Make all cards refactored and created dynamically with Javascript.
 -Remove commented code
-Add two remaining cards to be generated dynamically as others
-Add a hover effect dynamically to get the See Project button for all cards
-Fix eslint error

**Review by victorokoroji — APPROVED** [review]
Hi @Lul-Abdifan, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lul-Abdifan** [comment]
> Hi @Lul-Abdifan,
> # heavy_check_mark Status: APPROVED trophy tada confetti_ball
> 
> You have done a great job on this project, keep it up +1 1st_place_medal. Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: congratulations! tada100star2
> ## Top highlight dart
> 
>     * The Pull Request is following the GitHub Flow white_check_mark
> 
>     * The PR title and description are professional and descriptive white_check_mark
> 
>     * The commit messages are meaningful white_check_mark
> 
>     * Linters are working properly white_check_mark
> 
> 
>  [ 68747470733a2f2f6d65646961332e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966 ](https://camo.githubusercontent.com/5d2ef7789f70f7829b4c278f18e77bc67da5412b06f77ddc5b0c1b0cff45702a/68747470733a2f2f6d65646961332e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966) [ ](https://camo.githubusercontent.com/5d2ef7789f70f7829b4c278f18e77bc67da5412b06f77ddc5b0c1b0cff45702a/68747470733a2f2f6d65646961332e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966)
> ## Optional suggestions
> 
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!clapclapclap
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lul-Abdifan** [comment]
@[vickymarz](https://github.com/vickymarz) Thank you .:smile: Happy Coding.

**Pull request merged**